sql/sql_select.cc

```diff
@@ -27865,6 +27865,8 @@ void st_select_lex::print(THD *thd, String *str, enum_query_type query_type)
       }
     }
     str->append(" */ ");
+    if (join && join->cleaned)
+      return;
   }
```

Comment:
Please add a testcase to the patch.

An interesting observation is that the "Verify we're writing valid JSON" feature didn't catch this.  Apparently it only checks the JSON object/array structures, but not the strings...  if we get any other cases like this, it might make sense to add verification..  but this is outside of the scope of this MDEV.

Reply:
Added MDEV-30650 for future use.  Are we sticking to standard ASCII (space to ~) for these strings?

Reply:
Good. 
No, it is Unicode.  Currently UTF8mb3:
```
MariaDB [test]> show create table information_schema.optimizer_trace\G
*************************** 1. row ***************************
       Table: OPTIMIZER_TRACE
Create Table: CREATE TEMPORARY TABLE `OPTIMIZER_TRACE` (
  `QUERY` longtext NOT NULL,
  `TRACE` longtext NOT NULL,
  `MISSING_BYTES_BEYOND_MAX_MEM_SIZE` int(20) NOT NULL,
  `INSUFFICIENT_PRIVILEGES` tinyint(1) NOT NULL
) ENGINE=Aria DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci PAGE_CHECKSUM=0
1 row in set (0.004 sec)
```
(maybe we should change that to be UTF8MB4 in the next versions...)

Some characters need to be escaped, according to https://www.json.org/json-en.html, see the production for String.